jdaviz/configs/default/plugins/line_lists/tests/test_line_lists.py

```diff
@@ -141,3 +142,61 @@ def test_line_identify(specviz_helper, spectrum1d):
     ll_plugin.vue_change_visible(('Test List', line, 0))
     assert line.get('show') is False
     assert line.get('identify', False) is False
+
+
+def test_global_redshift_applied(specviz_helper, spectrum1d):
+    spec = Spectrum1D(flux=np.random.rand(100)*u.Jy,
+                      spectral_axis=np.arange(6000, 7000, 10)*u.AA)
+    specviz_helper.load_data(spec)
+
+    # Create a table with redshift included
+    lt = QTable()
+    lt['linename'] = ['O III', 'Halpha']
+    lt['rest'] = [5000., 6000.]*u.AA
+    lt['redshift'] = u.Quantity([0.0, 0.0])
+
+    with pytest.warns(UserWarning, match='per line/list redshifts not supported, use viz.set_redshift'):  # noqa
+        specviz_helper.load_line_list(lt)
+
+    # Load a line, and apply redshift globally
+    specviz_helper.plot_spectral_line("Halpha")
+    specviz_helper.set_redshift(0.01)
+    # Load second line, redshift should also be applied to it
+    specviz_helper.plot_spectral_line("O III")
+
+    viewer_lines = [mark for mark in specviz_helper.app.get_viewer(
+        specviz_helper._default_spectrum_viewer_reference_name).figure.marks
+        if isinstance(mark, SpectralLine)]
+
+    assert viewer_lines[0].redshift == 0.01
+    assert viewer_lines[1].redshift == 0.01
+
+
+def test_global_redshift_applied_to_all(specviz_helper, spectrum1d):
+    spec = Spectrum1D(flux=np.random.rand(100)*u.Jy,
+                      spectral_axis=np.arange(6000, 7000, 10)*u.AA)
+    specviz_helper.load_data(spec)
+
+    # Create a table with redshift included
+    lt = QTable()
+    lt['linename'] = ['O III', 'Halpha', 'O III']
+    lt['rest'] = [5000., 6000., 7000.]*u.AA
+    lt['redshift'] = u.Quantity([0.0, 0.0, 0.0])
+
+    with pytest.warns(UserWarning, match='per line/list redshifts not supported, use viz.set_redshift'):  # noqa
+        specviz_helper.load_line_list(lt)
+
+    # Load a line, so we can apply redshift
+    specviz_helper.plot_spectral_line("Halpha")
+    global_redshift = 0.01
+    specviz_helper.set_redshift(global_redshift)
+    # Load remaining lines
+    specviz_helper.plot_spectral_lines(global_redshift)
+
+    viewer_lines = [mark for mark in specviz_helper.app.get_viewer(
```

Comment:
If anyone has a suggestion to get around using a loop here and in the second test. It is here to get the SpectralLines in the viewer, and make sure the redshift is applied. The Qtable doesn't update automatically if redshift is applied so I went with this methodology for now.